### PR TITLE
Fix `name` and `materialIndex` causing crash on `.obj` meshes

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**
Fixes https://github.com/cyberbotics/webots/issues/4313

Although these fields should only apply to collada files, no distinction was made if the mesh happened to be an `obj`, resulting in crashes.

### CODE REVIEW

File: src/webots/nodes/WbMesh.cpp [hunk 1]
```diff
@@ -304,8 +308,9 @@ void WbMesh::updateTriangleMesh(bool issueWarnings) {
 }
 
 uint64_t WbMesh::computeHash() const {
-  const QByteArray meshPathNameIndex = (path() + mName->value() + mMaterialIndex->value()).toUtf8();
-  return WbTriangleMeshCache::sipHash13x(meshPathNameIndex.constData(), meshPathNameIndex.size());
+  QString meshPathNameIndex = path();
+  meshPathNameIndex += (mIsCollada ? mName->value() + QString::number(mMaterialIndex->value()) : "");
+  return WbTriangleMeshCache::sipHash13x(meshPathNameIndex.toUtf8().constData(), meshPathNameIndex.size());
```
Comment (BenjaminHug8) [inline src/webots/nodes/WbMesh.cpp:313]: Can we do this instead?

```suggestion
  const QByteArray meshHash = (path() + (mIsCollada ? mName->value() + mMaterialIndex->value()) : "").toUtf8();
  return WbTriangleMeshCache::sipHash13x(meshHash.constData(), meshHash.size());
```

File: src/webots/nodes/WbMesh.cpp [hunk 2]
```diff
@@ -97,6 +98,9 @@ void WbMesh::createResizeManipulator() {
 }
 
 bool WbMesh::checkIfNameExists(const aiScene *scene, const QString &name) const {
+  if (!mIsCollada)
+    return false;
+
```
Comment (BenjaminHug8) [inline src/webots/nodes/WbMesh.cpp:103]: Useless code, here is why:

`checkIfNameExists()` is called only if `mIsCollada` is `true` on line 161, so why check if it is false?

```suggestion
```

File: src/webots/nodes/WbMesh.cpp [hunk 3]
```diff
@@ -518,6 +523,8 @@ void WbMesh::updateUrl() {
   if (path().isEmpty())
     return;
 
+  mIsCollada = path().mid(path().lastIndexOf('.') + 1).toLower() == "dae" ? true : false;
```
Comment (omichel) [inline src/webots/nodes/WbMesh.cpp:526]: Simpler:
```suggestion
  mIsCollada = (path().mid(path().lastIndexOf('.') + 1).toLower() == "dae");
```

File: src/webots/nodes/WbMesh.cpp [hunk 4]
```diff
@@ -304,8 +308,9 @@ void WbMesh::updateTriangleMesh(bool issueWarnings) {
 }
 
 uint64_t WbMesh::computeHash() const {
-  const QByteArray meshPathNameIndex = (path() + mName->value() + mMaterialIndex->value()).toUtf8();
-  return WbTriangleMeshCache::sipHash13x(meshPathNameIndex.constData(), meshPathNameIndex.size());
+  QString meshPathNameIndex = path();
+  meshPathNameIndex += (mIsCollada ? mName->value() + QString::number(mMaterialIndex->value()) : "");
```
Comment (omichel) [inline src/webots/nodes/WbMesh.cpp:312]: Simpler:
```suggestion
  const QString meshPathNameIndex = path() + (mIsCollada ? mName->value() + QString::number(mMaterialIndex->value()) : "");
```

**Review by BenjaminHug8 — CHANGES_REQUESTED** [review]
Thank you, overall it is a good check but I have minor suggestions.

EDIT:
@omichel has been faster than me, so maybe some of my suggestions are not anymore valid.

File: src/webots/nodes/WbMesh.cpp (hunk 1 above)
Comment (BenjaminHug8) [reply src/webots/nodes/WbMesh.cpp:313]: Another suggestion above is better that mine.

File: src/webots/nodes/WbMesh.cpp (hunk 2 above)
Comment (ad-daniel) [reply src/webots/nodes/WbMesh.cpp:103]: Good point, thank you!

**Review by BenjaminHug8 — APPROVED** [review]
All good for me, thanks!

**Comment by ad-daniel** [comment]
they passed yes, I've just added a label and the checks restarted

**Review by omichel — APPROVED** [review]

**Pull request merged**